mondrian/src/main/java/mondrian/olap/fun/CrossJoinFunDef.java

```diff
@@ -999,29 +999,27 @@ private static boolean checkData(
         Evaluator evaluator)
     {
         if (cnt < 0) {
-            // no measures found, use standard algorithm
-            if (measureSet.isEmpty()) {
-                Object value = evaluator.evaluateCurrent();
-                if (value != null
-                    && !(value instanceof Throwable))
-                {
-                    return true;
-                }
-            } else {
+            boolean found = false;
+            Object value = evaluator.evaluateCurrent();
+            if (value != null
+                && !(value instanceof Throwable))
+            {
+                found = true;
+            }
+            if (!measureSet.isEmpty()) {
```

Comment:
So here's a possible fix:  https://github.com/mkambol/mondrian/commit/39fbd40187cd5dd646dfe206f1f274641f0b1064

The crossjoin optimizer cannot safely eliminate tuples with calc members, so simply skip them.  If tuples with calcs end up having empty results they should still be filtered out of the final results downstream.

@lucboudreau - curious about your opinion on this one.